ReleaseProcessAutomation/Jira/ServiceFacadeImplementations/JiraIssueService.cs

```diff
@@ -75,10 +75,24 @@ public IEnumerable<JiraToBeMovedIssue> FindAllNonClosedIssues (string versionId)
   public IEnumerable<JiraToBeMovedIssue> FindAllClosedIssues (string versionId)
   {
     var jql = $"fixVersion={versionId} and resolution != \"unresolved\"";
-    var resource = $"search?jql={jql}&fields=id,fixVersions";
+    var resource = $"search?jql={jql}&fields=id,fixVersions,summary";
     var request = _jiraRestClientProvider.GetJiraRestClient().CreateRestRequest(resource, Method.GET);
 
     var response = _jiraRestClientProvider.GetJiraRestClient().DoRequest<JiraNonClosedIssues>(request, HttpStatusCode.OK);
     return response.Data.Issues ?? throw new InvalidOperationException($"Could not get closed issue data from jira with request '{resource}'");
   }
+
+  public IEnumerable<JiraToBeMovedIssue> FindIssuesWithOnlyExactFixVersion (
+      HashSet<JiraProjectVersion> allVersions,
+      JiraProjectVersion exactFixVersion)
+  {
+    allVersions.RemoveWhere(projectVersion => projectVersion.id.Equals(exactFixVersion.id));
+
+    var allClosedIssues = FindAllClosedIssues(exactFixVersion.id);
+    var filteredIssues = allClosedIssues.Where(
+        i => i.Fields.FixVersions.TrueForAll(
+            v => !allVersions.Select(c => c.id).Contains(v.ID)));
+
+    return filteredIssues;
+  }
```

Comment:
The way we use the `HashSet<JiraProjectVersion>` does not make any sense. We are using it like a list therefore making it worse than if we used just a `List<JiraProjectVersion>`. There are multiple solutions possible how would you solve the problem?

Reply:
As im not that great with hashsets and new to LINQ i dont have a clue other than just removing the hashset and using a list altogether, but I dont think that that's the point ;)